src/pages/Reserve.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { useEffect, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { returnCurrentUser } from '../services/cookie';
+import { fetchCars } from '../redux/xers/carXer';
+import { addReservation } from '../redux/xers/reservationXer';
+import '../assets/App.css';
+import '../assets/reserve.css';
+
+const Reserve = () => {
+  const cars = useSelector((state) => state.cars);
+  const [city, setCity] = useState('');
+  const [date, setDate] = useState('');
+  const [car, setCar] = useState('');
+  const [errMsg, setErrMsg] = useState(null);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchCars());
+  }, [dispatch]);
+
+  useEffect(() => {
+    setErrMsg('');
+  }, [city, date, car])
+
+  const handleMakeReservation = (e) => {
+    e.preventDefault();
+    const data = [{
+      car_id: car, date, city,
+    }, returnCurrentUser().id];
+    if (city === '' || date === '' || car === '') {
+      setErrMsg('All fields are required');
+      return;
+    }
+    dispatch(addReservation(data));
+    window.location.href = '/reservations';
+  };
+
+  return (
+    <div className="content">
+      <h3 className="reserve-title">Reservation form</h3>
+      <div className="reserve-form">
+        <form className="form">
+          <div className="input-container">
+            {' '}
+            <input type="text" placeholder="Enter Your Username..." value={returnCurrentUser().username} disabled />
+          </div>
+          <div className="input-container">
+            <select className="select-input" onChange={(e) => setCar(e.target.value)}>
+              <option value=""> Select modal</option>
```

Comment:
- [ ] This value should be "Select model", not "Select modal".  Kindly fix this typo. 👍